src/ape/api/providers.py

```diff
@@ -1046,7 +1047,13 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
     def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         try:
-            txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            account = self.account_manager[txn.sender]
+            is_contract = self.web3.eth.get_code(txn.sender) != "0x"
```

Comment:
`web3.eth.get_code()` returns `HexBytes`, and therefore it will never be equal to `"0x"` (the str).
Using `is_contract` as @fubuloubu said would fix that thought!